test/python/pulse/test_builder.py

```diff
@@ -1229,3 +1229,324 @@ def test_call_subroutine_with_parametrized_duration(self):
             pulse.call(subroutine)
 
         self.assertEqual(len(main.blocks), 1)
+
+
+class TestSubroutineWithCXGate(QiskitTestCase):
```

Comment:
I think this test should cover the case where the name of a parameter has a `.` in it.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8005/commits/b4a90adf082b691dc67851d2837c2b45d6ea0953